### ISSUE TITLE
[Bug Report] ttnn.remainder op gives nan after recent sfpu-float-32 commit from Dec11 causing tt-forge test_remainder fail

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-metal#16011]
(Opening bug for PR/tracking purposes, issue mostly understood now, flagged yesterday in slack - thanks @KalaivaniMCW )

**Describe the bug**

The tt-forge-fe test called test_remainder is [failing](https://github.com/tenstorrent/tt-forge-fe/actions/runs/12303027711/job/34338812052) with incorrect output (nans) after tt-metal uplift on Dec12, I bisected it to be caused by tt-metal commit b959916b5a which wasn't too difficult since we were tracking tt-metal daily.

I don't have good tt-forge-fe repro results (never tried it locally just on CI) aside from:

- Checkout tt-forge-fe at commit 74546e658a (contains just this tt-metal change that causes problem) 
- and build according to instructions https://docs.tenstorrent.com/tt-forge-fe/build.html
- run `pytest forge/test/mlir/test_ops.py::test_remainder`

Better to reproduce in tt-metal directly next section:

**To Reproduce**

ttnn repro result are easier and working already - put this in a python file called `test_remainder.py` and run it with `python test_remainder.py`

```
import torch
import ttnn

with ttnn.manage_device(0) as device:
    input1 = torch.randn(2, 32, 32)
    input2 = torch.randn(2, 32, 32)
    torch_output = torch.remainder(input1, input2)

    input1 = ttnn.from_torch(input1, dtype=ttnn.float32)
    input2 = ttnn.from_torch(input2, dtype=ttnn.float32)

    input1 = ttnn.to_device(input1, device, memory_config=ttnn.DRAM_MEMORY_CONFIG)
    input2 = ttnn.to_device(input2, device, memory_config=ttnn.DRAM_MEMORY_CONFIG)

    input1 = ttnn.to_layout(input1, ttnn.TILE_LAYOUT)
    input2 = ttnn.to_layout(input2, ttnn.TILE_LAYOUT)

    output = ttnn.remainder(input1, input2)

    output = ttnn.to_torch(output)
    print(output)
    print(torch_output)
```

Output:

```
TorchTensor([[[nan, nan, nan,  ..., nan, nan, nan],
              [nan, nan, nan,  ..., nan, nan, nan],
              [nan, nan, nan,  ..., nan, nan, nan],
              ...,
<snip>
tensor([[[ 0.7699,  0.6809,  0.0599,  ..., -0.2512, -0.0408,  0.0277],
         [ 0.3033,  0.6022, -0.2856,  ..., -1.2751,  0.0034,  1.2214],
         [-0.1675,  0.7429,  0.0169,  ..., -0.1604, -0.5683, -0.0239],
         ...,
```

Gives Nans at b959916b5a, passes one commit earlier at 2f59d5e02e .  Also passes with bfloat16 instead of float32.

```
024-12-11 (b959916b5a) by GitHub (Author 156762498+KalaivaniMCW@users.noreply.github.com) : #15483: Binary sfpu ops for Float32 precision (#15805)
2024-12-10 (2f59d5e02e) by GitHub (Author afuller@tenstorrent.com) : Clang-tidy incremental scan for branches (#15888)
```

**Expected behavior**

Good looking output, not Nans

**Comment by KalaivaniMCW** [comment]
Related to #15780 issue 
The above test is passing on my working branch https://github.com/tenstorrent/tt-metal/pull/15992 
will update once it is ready 

<img width="1512" alt="Image" src="https://github.com/user-attachments/assets/ff185a04-c6be-4d8a-89f0-582520c9165e" />

**Comment by eyonland** [comment]
Thank you for the PR @KalaivaniMCW!

**Comment by kmabeeTT** [comment]
I see #15992 was merged (thank you), and I pulled latest tt-metal commit 388e187b27f8924ea09512719f666407c40f6bd9 at that merge into tt-mlir (released it just now) after testing against tt-forge-fe that original remainder op is now [passing](https://github.com/tenstorrent/tt-forge-fe/actions/runs/12356654654). Going to mark this closed.
